File: roman/validation.py
from roman.convert import dict_num

def limit(input):
	count = 1
	for i in range(1, len(input)) :
		if input[i] == input[i-1] :
			count+= 1 
			if count > 3 :
				return False
		else:
			count = 1
	return True

def sintaxis(input_str):
    for i in range(len(input_str) - 1):
        current_char = input_str[i]
        next_char = input_str[i + 1]

        current_value = dict_num[current_char]
        next_value = dict_num[next_char]

        if current_value < next_value:
            if not (
                (current_char == "I" and next_char in ("V", "X")) or
                (current_char == "X" and next_char in ("L", "C")) or
                (current_char == "C" and next_char in ("D", "M"))
            ):
                print("esto no se puede:", current_char + next_char)
                return False
    return True

File: go_code/romanos/validation.go
package romanos

import "fmt"

var romanValues = map[string]int{
	"I": 1,
	"V": 5,
	"X": 10,
	"L": 50,
	"C": 100,
	"D": 500,
	"M": 1000,
}

// funcion para delimitar el mismo caractar a no mas de 3 repeticiones
func TripleRestriccion(input string) bool {
	count := 1
	for i := 1; i < len(input); i++ {
		if input[i] == input[i-1] {
			count++
			if count > 3 {
				return false
			}
		} else {
			count = 1
		}
	}

	return true
}

// funcion para decidir que puede restar y q no
func Sintaxis(input string) bool {
	for i := 0; i < len(input)-1; i++ {
		currentChar := string(input[i])
		nextChar := string(input[i+1])

		currentValue := romanValues[currentChar]
		nextValue := romanValues[nextChar]

		if currentValue < nextValue {
			// cosas que SI se pueden restar
			if !(currentChar == "I" && (nextChar == "V" || nextChar == "X")) &&
				!(currentChar == "X" && (nextChar == "L" || nextChar == "C")) &&
				!(currentChar == "C" && (nextChar == "D" || nextChar == "M")) {
				fmt.Println("esto no se puede", currentChar+nextChar)
				return false
			}
		}
	}
	return true
}
